enumerable_methods.rb

```diff
@@ -0,0 +1,132 @@
+module Enumerable
+  # start of my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+  # end of my_each
+
+  # start of my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+  # end of my_each_with_index
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    new_array = []
+    i = 0
+
+    until i >= length
+      new_array << i if yield self[i]
+      i += 1
+    end
+  end
+  # end of my_select
+
+  # my_all?
+  def my_all?
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
+  # end of my_all?
+
+  # my_any?
+  def my_any?
+    my_each do |item|
+      return true if yield item
+    end
+    false
+  end
+  # end of my_any?
+
+  # my_none?
+  def my_none?
+    my_each do |item|
+      return false if yield item
+    end
+    true
+  end
+  # end of my_none?
+
+  # my_count
+  def my_count
+    count = 0
+    my_each do |_item|
+      count += 1
+    end
+    count
+  end
+  # end of my_count
+
+  # start of my_map
+  def my_map(_proc = nil)
+    new_array = []
+    my_each do |item|
+      new_array << item
+    end
+    new_array
+  end
+  # end of my_map
+
+  # start of my_inject
+  def my_inject(acc)
```

Comment:
- [ ] [`inject`](https://rubyapi.org/2.7/o/enumerable#method-i-inject) takes initial value to start with and it's optional, for example:
```ruby
# without passing initial value to inject
[5, 2, 1].inject { |result, x| result + x }
# with passing initial value to inject
[5, 2, 1].inject(0) { |result, x| result + x }
```

In case of not passing initial value, ruby will select the first item of the array to be the initial value. Your current implementation breaks if the initial value is not passed. Let's add a case if the initial value is not passed.